webapp/apps/btax/helpers.py

```diff
@@ -0,0 +1,262 @@
+                        self.min = self.min[1:]
+
+
+def get_btax_defaults():
+    from btax import DEFAULTS
+    defaults = dict(DEFAULTS)
+    for k,v in defaults.items():
+        v['col_label'] = ['']
+    BTAX_DEFAULTS = {}
+    for k in (BTAX_BITR + BTAX_OTHER + BTAX_ECON):
+        param = BTaxParam(k,defaults[k])
+        BTAX_DEFAULTS[param.nice_id] = param
+    for k in BTAX_DEPREC:
+        fields = ['{}_{}_Switch'.format(k, tag)
+                     for tag in ('gds', 'ads',  'tax')]
+        for field in fields:
+            param = BTaxParam(field, defaults[field])
+            BTAX_DEFAULTS[param.nice_id] = param
+        for field in ['{}_{}'.format(k, 'exp')]:
+            param = BTaxParam(field, defaults[field])
+            BTAX_DEFAULTS[param.nice_id] = param
+    return BTAX_DEFAULTS
+
+BTAX_DEFAULTS = get_btax_defaults()
+
+
+def group_args_to_btax_depr(btax_default_params, asset_yr_str):
+    depr_field_order = ('gds', 'ads', 'exp', 'tax')
+    depr_argument_groups = []
+    for yr in ['all'] + asset_yr_str:
+        gds_id = 'btax_depr_{}yr_gds_Switch'.format(yr)
+        ads_id = 'btax_depr_{}yr_ads_Switch'.format(yr)
+        tax_id = 'btax_depr_{}yr_tax_Switch'.format(yr)
+        exp_id = 'btax_depr_{}yr_exp'.format(yr)
+        radio_group_name = 'btax_depr_{}yr'.format(yr)
+        field3_cb = 'btax_depr_{}yr_exp'.format(yr)
+        field4_cb = 'btax_depr_{}yr_tax'.format(yr)
+
+        if yr == 'all':
+            pretty = "Check all"
+            is_check_all = True
+            label = ''
+            td_style_class = 'table-check-all-item'
+            tr_style_class = 'tr-check-all'
+        else:
+            td_style_class = ''
+            is_check_all = False
+            label = ''
+            tr_style_class = ''
+            pretty = '{}-year'.format(yr.replace('_', '.'))
+        depr_argument_groups.append(
+            dict(param1=btax_default_params[gds_id],
+                 field1=gds_id,
+                 param2=btax_default_params[ads_id],
+                 field2=ads_id,
+                 param3=btax_default_params[tax_id],
+                 field3=tax_id,
+                 param4=btax_default_params[exp_id],
+                 field4=exp_id,
+                 radio_group_name=radio_group_name,
+                 asset_yr_str=yr,
+                 tr_style_class=tr_style_class,
+                 pretty_asset_yr_str=pretty,
+                 field3_cb=field3_cb,
+                 field4_cb=field4_cb,
+                 is_check_all=is_check_all,
+                 label=label,
+                 td_style_class="table-check-all-item" if yr == 'all' else '')
+
+            )
+    return depr_argument_groups
+
+def btax_results_to_tables(results, first_budget_year):
+    """
+    Take various results from dropq, i.e. mY_dec, mX_bin, df_dec, etc
+    Return organized and labeled table results for display
+    """
+    asset_col_meta = dict(btax.parameters.DEFAULT_ASSET_COLS)
```

Comment:
also is this function similar enough to taxcalc_results_to_tables that we could use the old one? Might not be, just wondering.